### ISSUE TITLE
Please add support for Crashlytics

### ISSUE THREAD

**Issue by dotdoom** [flutter/flutter#14765]
In #9592 there's a request for adding Crashlytics, but it was blocked by an absence of crash reporting support in Flutter. Now that there are ways to intercept errors, it should be possible to add Crashlytics / Firebase error reporting.

**Comment by xster** [comment]
cc @collinjackson might have updates. 

I'd like to use it for the gallery as well #4070

**Comment by hocinehamdi** [comment]
It will be perfect to have first class dart crash reporting tool

**Comment by pulyaevskiy** [comment]
Very much interested in this as well. It would be great to have crash reports in Firebase console for both Dart and native side errors.

**Comment by PareshMayani** [comment]
Indeed it's much needed thing! Looking forward for either crashlytics or Firebase integration

**Comment by wahibhaq** [comment]
I had to use Crashlytics and ended up on this page. It would be really nice to have it for flutter 🙏

**Comment by jpgilchrist** [comment]
Given the latest updates in Google I/O, this seems like a pretty obvious thing to include. firebase_analytics plugin already provides the google analytics portion. However, the crashlytics isn't available, which is a major part of the firebase offerings.

**Comment by AppleEducate** [comment]
Definitely would love this plugin!

**Comment by eseidelGoogle** [comment]
Looks like this is now our top most requested issue. :D

@yjbanov looked at this briefly last year and determined it would require some changes to Crashlytics itself to add Dart support.  We'll reach out to the crashlytics team again and see how things have changed in the intervening months.  This seems like a very reasonable request and should definitely be doable.  Just a question of when. :)

**Comment by eseidelGoogle** [comment]
I spoke with @collinjackson a while back and it sounded like they or @kroikie or @bparrishMines were interested in taking this on.  I believe mostly this requires some coordination with the crashlytics team.  FWIW this is still our top-most requested bug. :)

**Comment by ChristBKK** [comment]
+1 for this one

**Comment by gonzalonm** [comment]
Any news on this? I would really appreciate to have Crashlytics integration on Flutter!

**Comment by vithanichandresh** [comment]
+1

**Comment by jjbeto** [comment]
+1

**Comment by jjbeto** [comment]
+1

**Comment by jaumard** [comment]
Any news on this ? Did the plugin start ? I want to take a look see if I can help. Would be nice to use Crashlytics (and other reporting tool of Fabric) with Flutter as we're using a lot of functionality of Fabric into our native app already.

**Comment by karantanwar** [comment]
it would be great to have this plugin

**Comment by jaumard** [comment]
Hello everyone, I've just implemented a plugin for Crashlytics integration for iOS and Android. Reports look like this: 
<img width="1374" alt="capture d ecran 2018-08-21 a 10 11 13" src="https://user-images.githubusercontent.com/2970449/44389224-931a8b80-a52a-11e8-9c16-ebab37a0dd31.png">

It's not perfect but at least it's a start :) it can also allow you to set custom logs and informations (general and user info) from Dart.

Let me know what you think of it guys. 

One limitation is that as the report is with Crashlytics SDK, it reports manual exceptions as Non Fatal, one way to bypass that would be to make the app crash, I've opened an issue on my repo to collect thoughts about this.

Here is the repo https://github.com/kiwi-bop/flutter_crashlytics and the pub packages https://pub.dartlang.org/packages/flutter_crashlytics

**Comment by nstrelow** [comment]
It would be awesome to get official support for Crashlytics!

**Comment by jaumard** [comment]
Hey @eseidelGoogle :) any news about your synchronization with Fabric team and an official support ? Would be nice to see some changes here, or at least make them change their API to allow reporting fatal crashes. 
Because the main issue on Crashlytics API is: 

- android `logException` only report non fatal crash, a boolean to be able to tag it crash would be nice
- iOS `recordCustomExceptionName` also report only non fatal crash, having also a boolean would be helpful

With this two small changes we can get a real Crashlytics support and we'll not be able to hack native exception to rebuild one from dart one.

**Comment by eseidelGoogle** [comment]
Thanks @jaumard.  I'm still working with that team (slowly).  I expect it will be a while until have anything to share though.  But thank you for all the work you've done so far (including the clear enumeration of needs)!

**Comment by Hixie** [comment]
@kroikie I've heard through the grapevine that you're working on this so I assigned it to you. Please update the milestone field to be the month you (very conservatively) think you'll be done. Thanks!
And thanks for working on this!

**Comment by neiljaywarner** [comment]
@Hixie @eseidelGoogle @kroikie Can we please get a recipe on flutter.io that does not require a sentry.io account?

Ideally with a link to a section somehow detailing pros and cons compared to the first-class support that is a work in progress.

It looks like crashlytics support is "good enough" right now but improving, but i can't figure out yet how to clarify that, and the need is increasing.  thank you

**Comment by Hixie** [comment]
@neiljaywarner are you asking for sample code that doesn't involve `sentry.io`?

**Comment by neiljaywarner** [comment]
Yes please

**Comment by hummatli** [comment]
I like the idea! Google must do this!

**Comment by GreenAppers** [comment]
Debugging my app's SIGSEGV in libflutter.so, I've been working to extend flutter_crashlytics to include CrashlyticsNdk and thereby actually catch SIGSEGV and report it.

However I have not been able to successfully symbolicate the crashes.

The problem seems to be that "flutter build" produces no artifacts with debugging symbols.  The files under build/app/intermediates/transforms/stripDebugSymbol and build/app/intermediates/transforms/mergeJniLibs are nearly same size, and it seems that none of them contain debugging symbols.

We need to modify "flutter build" to produce some libflutter.so with debugging symbols.  Then configure build.gradle like:

crashlytics {
    enableNdk true
    androidNdkOut "../../build/app/intermediates/transforms/mergeJniLibs/release/0/lib"
    androidNdkLibsOut "../../build/app/intermediates/transforms/stripDebugSymbol/release/0/lib"
}

And upload with ./gradlew crashlyticsUploadSymbolsRelease.

Many, if not all, flutter_crashlytics users need this.  As I've learned, writing a pure Dart app will not protect you from native crashes.  Without CrashlyticsNdk we can't learn of users' SIGSEGV, and without changes to "flutter build" we can't symbolicate them.

**Comment by eseidelGoogle** [comment]
https://github.com/flutter/flutter/wiki/Crashes includes instructions on how to get flutter.jar and flutter.framework files with symbols in them.  Let me know if that doesn't help?

**Comment by GreenAppers** [comment]
Thanks for this and the quick response.  This is exactly what I needed.

It would be nice to find an easier solution to getting these uploaded to Crashlytics.

**Comment by GreenAppers** [comment]
I would appreciate any comments on PR towards symbolicating native crashes in Flutter apps with Crashlytics: https://github.com/kiwi-bop/flutter_crashlytics/pull/63/files

Before:

![image](https://user-images.githubusercontent.com/47840244/54049678-0667b000-4192-11e9-88b2-5aaa997145c8.png)

After:

![image](https://user-images.githubusercontent.com/47840244/54049906-bc32fe80-4192-11e9-9c40-95f713a6c52d.png)

Until I can fix the bug I don't know if these frames are correct.  They seem plausible because this crash is preceded by video artifacts in canvas.

**Comment by MisterJimson** [comment]
Should this be closed? The pub link on the readme goes nowhere. Is the package published?

https://github.com/flutter/plugins/tree/master/packages/firebase_crashlytics

**Comment by jaumard** [comment]
Also notice that is a partial support, as it does not more than flutter_crashlytics and have same limitations... the main who is that crash are reported as non fatal :/

**Comment by kroikie** [comment]
@MisterJimson The [pub link](https://pub.dartlang.org/packages/firebase_crashlytics) should be working now.

@jaumard You are correct that the crashes are reported as non fatal. Future versions will address this, thanks for the feedback.

**Comment by kroikie** [comment]
Filed an [issue](https://github.com/flutter/flutter/issues/29689) for fatal support

**Comment by kroikie** [comment]
@dotdoom

This issue has been moved to https://github.com/FirebaseExtended/flutterfire/issues/903. Any further collaboration will be done there.
